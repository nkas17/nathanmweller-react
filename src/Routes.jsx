import React from 'react';
import { HashRouter, Route } from 'react-router-dom';
import HomePage from './pages/HomePage';
import FedPage from './pages/FedPage';
import BlogPage from './pages/BlogPage';
import AboutPage from './pages/AboutPage';

/**
 * Main component that holds all routes
 */
/* eslint-disable-next-line react/prefer-stateless-function */
class Routes extends React.Component {
	render() {
		return (
			<HashRouter>
				<main>
					<Route exact path="/" component={HomePage} />
					<Route exact path="/fed" component={FedPage} />
					<Route path="/blog" component={BlogPage} />
					<Route exact path="/about" component={AboutPage} />
				</main>
			</HashRouter>
		);
	}
}

export default Routes;
