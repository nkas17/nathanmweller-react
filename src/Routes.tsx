import React from 'react';
import { HashRouter, Route } from 'react-router-dom';
import FedPage from './pages/FedPage';
import BlogPage from './pages/BlogPage';
import AboutPage from './pages/AboutPage';
import Header from './components/Header';

/**
 * Main component that holds all routes
 */
function Routes() {
  return (
    <HashRouter>
      <main>
        <Route exact path="/" component={() => <Header alwaysShow />} />
        <Route exact path="/fed" component={FedPage} />
        <Route path="/blog" component={BlogPage} />
        <Route exact path="/about" component={AboutPage} />
      </main>
    </HashRouter>
  );
}

export default Routes;
