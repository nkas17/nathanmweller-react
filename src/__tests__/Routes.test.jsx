import React from 'react';
import renderer from 'react-test-renderer';
import Routes from '../Routes';

it('renders correctly', () => {
	const tree = renderer.create(<Routes />).toJSON();
	expect(tree).toMatchSnapshot();
});
