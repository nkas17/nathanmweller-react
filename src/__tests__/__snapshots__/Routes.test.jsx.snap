// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <header
    className="page-header flex-wrapper"
  >
    <div
      className="personal-info page-header__flex-item flex-item"
    >
      <h1
        className="page-header__title"
      >
        nathan m weller
      </h1>
      <p
        className="page-header__description page-header__description--show"
      >
        husband, father, software architect, web developer, cook, photographer, f√∫tbol player, gamer
      </p>
      <ul
        className="contact-info page-header__contact-info"
      >
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="mailto:nathan@nathanmweller.com?subject=Reaching%20Out"
          >
            <i
              className="fa fa-envelope-o"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://www.linkedin.com/pub/nathan-weller/52/b05/552"
          >
            <i
              className="fa fa-linkedin"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://twitter.com/nkas17_m"
          >
            <i
              className="fa fa-twitter"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://plus.google.com/u/0/100022763393186980494/posts/p/pub"
          >
            <i
              className="fa fa-google-plus"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://www.facebook.com/nathan.weller.9"
          >
            <i
              className="fa fa-facebook"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="tel:6168430170"
          >
            <i
              className="fa fa-phone"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="sms:6168430170"
          >
            <i
              className="fa fa-comment-o"
            />
          </a>
        </li>
      </ul>
    </div>
    <nav
      className="nav flex-item"
    >
      <ul>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="#/"
            onClick={[Function]}
          >
            home
          </a>
        </li>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="#/fed"
            onClick={[Function]}
          >
            fed
          </a>
        </li>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="https://www.blog.nathanmweller.com"
          >
            blog
          </a>
        </li>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="#/about"
            onClick={[Function]}
          >
            about
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="logo flex-item logo--show"
    >
      <img
        alt="nathan m weller"
        className="logo__image"
        height="200"
        src="test-file-stub"
        width="200"
      />
    </div>
  </header>
</main>
`;
